solutions/convert_string_to_uppercase.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+A module for converting any given string to uppercase.
+Module contents:
+    - convert_string_to_uppercase: A function that converts any given string to uppercase.
+Created on 29 Dec 2024
+@author: Noorelsalam Almakki
+"""
+
+
+def convert_string_to_uppercase(text: str) -> str:
+    """The convert_string_to_uppercase function takes a string and returns the uppercase version of it.
+
+    Parameters:
+        - text (str): any string text.
+    Returns:
+        - upper_string (str): the uppercase version of the input text.
+    Raises:
+        - AssertionError: if the input text is not a string.
+    Example:
+    >>> convert_string_to_uppercase("hello")
+    'HELLO'
+    >>> sum_of_digits("HELLo")
```

Comment:
The doctests have errors. The reference to `sum_of_digits` is incorrect. Here is the correct function:
```suggestion
    >>> convert_string_to_uppercase("HELLo")
```

Reply:
There must be some kind of confusion. This pull request is for "convert a string to uppercase" not the "sum of digits" challenge.

Reply:
Yes, i am aware. Hence, it's why  you should replace `sum_of_digits("HELLo")` function in your code with `convert_string_to_uppercase("HELLo")`

Reply:
Ohhh, good catch! I will modify it. Thanks alot